### PR TITLE
Enable Regression Analysis

### PR DESCRIPTION
**Pull request by camagallen**
## Overview
- Enables regression versions of reporting functions, with basic measures added for the regression version of bias_report.
- Adds tutorial, template, and example notebooks for regression features
- Improves management of significant figures as required to enable flagging of regression outputs
- Minor update to the PR template

Resolves: [AB#10686](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/10686), [AB#16154](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/16154), [AB#10683](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/10683)

## Checklist
- [X] I have read the [Contributing](https://github.com/KenSciResearch/docs/code_contributions/CONTRIBUTING.md) guidelines
- [x] I performed a self-review of my own code
- [x] The CHANGELOG and version.json have been updated
- [x] Documentation has been updated to reflect the changes
    - [x] The code has clear and helpful comments, particularly in hard-to-understand areas
    - [x] READMEs, Tutorials/Examples, and links have been updated as appropriate
- [X] Tests have been added to prove that the fix is effective or that the feature works
- [X] Pytest was run; new and existing unit tests pass locally with my changes
- [X] I have made note of any breakages or dependencies (if applicable)
- [X] Any dependent changes have been merged and published in downstream modules
- [X] These changes generate no new warnings

## How was this update tested?
pytest and visual inspection of new notebooks

### CODE REVIEW

File: tutorials_and_examples/Example-ToolUsage_Regression.ipynb [hunk 1]
```diff
@@ -0,0 +1,1287 @@
+{
```
Comment (kenviku) [inline tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: <p>Given that one value of feature is compared to every-other value of the same feature, I believe it might be helpful if we separate the tables for each feature! That is <code>gender</code>  has its own table report followed by <code>col2</code>  with its own table. In the above format, its very hard to follow the numbers otherwise. </p><p> </p><p>For our discussion: I believe given the above table - a heatmap of rows and columns of values with cells representing one metric at a time could be easier to interpret! </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/KenSciResearch/fairMLHealth/pull/48/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (kenviku) [inline tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: <p>For cohort evaluations, isnt the report need to be aggregated at each quantile level? That is quantile = 0; Gender = [0,1]. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/KenSciResearch/fairMLHealth/pull/48/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (camagallen) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: <p>@viku couldn't we leave it to the user to submit multiple requests if they would like multiple tables?</p>

Comment (camagallen) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: <p>@kenviku could you rephrase? I'm not able to visualize what you're asking for.</p>

Comment (kenviku) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: Yes but a better and easy-to-interpret outcome shall be functionality (and responsibility) of the library. Will lead to better standardization and consistency across diverse usage!

Comment (kenviku) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: In current format, a metric (lets say MAE ratio) is defined for groups of [quantile, col2, gender] - where quantile is derived from `col2` values. Isn't it better for the metric to be defined at group of [quantile, gender] only? In this case, one can compare - what is the metric value of one-quantile with gender X compared to other-quantiles with gender X?

Comment (camagallen) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: Will take this offline to clarify.

Comment (camagallen) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: Will take this offline to clarify.

Comment (kenviku) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: The comment is captured in the Issue #50 to be addressed in follow-up PRs.

Comment (kenviku) [reply tutorials_and_examples/Example-ToolUsage_Regression.ipynb:1]: Captured in Issue #51 post offline discussion.

**Review by kenviku — APPROVED** [review]
LGTM with few open issues to be addressed in follow-up PRs.
